File: Recursion/palindrome.c
#include <stdio.h>

int isPalindrome(int num, int temp) {
    if (num == 0)
        return temp;
    else {
        temp = temp * 10 + num % 10;
        return isPalindrome(num / 10, temp);
    }
}

int main() {
    int number;
    printf("Enter a number: ");
    scanf("%d", &number);
    int reversed = isPalindrome(number, 0);
    if (number == reversed)
        printf("%d is a palindrome.\n", number);
    else
        printf("%d is not a palindrome.\n", number);
    return 0;
}

File: Recursion/palindrome.py
def is_palindrome(num, temp):
    if num == 0:
        return temp
    else:
        temp = temp * 10 + num % 10
        return is_palindrome(num // 10, temp)

number = int(input("Enter a number: "))
reversed_num = is_palindrome(number, 0)
if number == reversed_num:
    print(number, "is a palindrome.")
else:
    print(number, "is not a palindrome.")
